### ISSUE TITLE
False negative with incompatible properties in subclass with multiple parents

### ISSUE THREAD

**Issue by robert-bn** [microsoft/pyright#8167]
Consider the following example:
```python
class HasInt:
    @property
    def value(self) -> int:
        return 3

class HasStr:
    @property
    def value(self) -> str:
        return "three"

class HasBoth(HasInt, HasStr): ...
```
`HasInt` and `HasStr` are incompatible with each other, but pyright doesn't see an issue.
If we examine the type of the `value` field in a `HasBoth` instance, we see that it resolves to `int`
```python
def example(x: HasBoth):
    reveal_type(x.value) # information: Type of "x.value" is "int"
```
This may be correct with the python method resolution order, but it violates the Liskov substitution principle: any function that is expecting a `HasStr` instance is expecting a `str` for the value property. 

It's easy to produce a type error at runtime using this, for instance:
```python
def length_of_value(x: HasStr):
    len(x.value)

length_of_value(HasBoth())
````

Note that mypy reports an issue with the `HasBoth` class:
```
Definition of "value" in base class "HasInt" is incompatible with definition in base class "HasStr"
```

In addition, if one removes the `@property` decorator, pyright does complain that the base classes are incompatible. 

Tested with pyright version 1.1.367

**Comment by erictraut** [comment]
Thanks for the bug report. As you noted, property validation was missing in the multi-inheritance case. I've added the missing functionality, and it will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.369.
